wire-ios-data-model/Source/E2EIdentity/GracePeriodRepository.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public protocol GracePeriodRepositoryInterface {
```

Comment:
I haven't fully understood why we need a `GracePeriodRepository`. Don't we have the certificate from which we can retrieve the expiration date?

Reply:
`GracePeriod` is the period during which we must get the certificate.

Reply:
I will add documentation.

Reply:
There is an explanation at line 28

Reply:
https://github.com/wireapp/wire-ios/pull/1040

Reply:
add documentation